test/addons_test.go

```diff
@@ -366,24 +515,34 @@ func testgroup(t *testing.T, groupname string, version string, jobs ...clusterTe
 			return err
 		}
 
-		t.Logf("found old version of addon %s %s (revision %s) and new version %s (revision %s)", newAddon.GetName(), oldRev, oldVersion, newVersion, newRev)
-		if oldVersion.GT(newVersion) {
-			return fmt.Errorf("revisions for addon %s are broken, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
-		}
-		if !newVersion.GT(oldVersion) {
+		if newVersion.EQ(oldVersion) {
 			t.Logf("skipping upgrade test for addon %s, it has not been updated", newAddon.GetName())
+			addonDeploymentsArray = append(addonDeploymentsArray, oldAddon)
```

Comment:
same comment here: if no upgrade test needed, we probably don't need to actually do anything more for the addon right? this would essentially rerun `testGroupDeployment` which is just deploying the latest addon version